tests/dns.rs

```diff
@@ -0,0 +1,90 @@
+use coinswap::{
+    maker::error::MakerError,
+    market::rpc::{RpcMsgReq, RpcMsgResp},
+    utill::{read_message, send_message},
+};
+use std::{
+    fs, io,
+    io::BufRead,
+    net::TcpStream,
+    path::PathBuf,
+    process::{Child, Command},
+    sync::mpsc::{self, Receiver, Sender},
+    thread,
+    time::Duration,
+};
+
+fn spawn_directoryd_thread(log_sender: Sender<String>) -> Child {
+    let mut directoryd_process = Command::new("cargo")
+        .args(["run", "--bin", "directoryd"])
+        .stdout(std::process::Stdio::piped())
+        .spawn()
+        .expect("Failed to start directoryd process");
+
+    let stdout = directoryd_process.stdout.take().unwrap();
+    thread::spawn(move || {
+        let reader = io::BufReader::new(stdout);
+        for log_message in reader.lines().map_while(Result::ok) {
+            log_sender.send(log_message).unwrap();
+        }
+    });
+    directoryd_process
+}
+
+fn send_rpc_req(req: &RpcMsgReq) -> Result<Option<RpcMsgResp>, MakerError> {
+    let mut stream = TcpStream::connect("127.0.0.1:4321")?;
+    stream.set_read_timeout(Some(Duration::from_secs(20)))?;
+    stream.set_write_timeout(Some(Duration::from_secs(20)))?;
+
+    send_message(&mut stream, req)?;
+
+    let resp_bytes = read_message(&mut stream)?;
+    let resp: RpcMsgResp = serde_cbor::from_slice(&resp_bytes)?;
+    Ok(Some(resp))
+}
```

Comment:
No we need to send the RPC request using `directory-cli` app commands. Similar to running the `directoryd` app with `std::process::Command`. 

Then only it will test the cli app. Now we are just doing vanilla network call. That's not testing anything.